carma-messenger-core/truck_inspection_plugin/src/truck_inspection_plugin_node.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2022 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#include "truck_inspection_plugin/truck_inspection_plugin_node.hpp"
+
+namespace truck_inspection_plugin
+{
+  namespace std_ph = std::placeholders;
+
+  Node::Node(const rclcpp::NodeOptions &options)
+      : carma_ros2_utils::CarmaLifecycleNode(options)
+  {
+    // Create initial config
+    config_ = Config();
+
+    // Declare parameters
+    config_.num_of_entries = declare_parameter<int>("num_of_entries", config_.num_of_entries);
+  }
+
+  rcl_interfaces::msg::SetParametersResult Node::parameter_update_callback(const std::vector<rclcpp::Parameter> &parameters)
+  {
+    // TODO for the USER: Ensure all parameters can be updated dynamically by adding them to this method
+    auto error = update_params<int>({{"num_of_entries", config_.num_of_entries}}, parameters);
+
+    rcl_interfaces::msg::SetParametersResult result;
+
+    result.successful = !error;
+
+    return result;
+  }
+
+  carma_ros2_utils::CallbackReturn Node::handle_on_configure(const rclcpp_lifecycle::State &)
+  {
+    // Reset config
+    config_ = Config();
+
+    // Load parameters
+    get_parameter<int>("num_of_entries", config_.num_of_entries);
+
+    // Register runtime parameter update callback
+    add_on_set_parameters_callback(std::bind(&Node::parameter_update_callback, this, std_ph::_1));
+
+    // Setup subscribers
+    mo_sub_ = create_subscription<carma_v2x_msgs::msg::MobilityOperation>("mobility_operation_inbound", 5,
+                                                              std::bind(&Node::mobilityOperationCallback, this, std_ph::_1));
+
+    // Setup publishers
+    mr_pub_ = create_publisher<carma_v2x_msgs::msg::MobilityRequest>("mobility_request_outbound", 5);
+    cav_detection_pub_ = create_publisher<std_msgs::msg::String>("cav_truck_identified", 5);
+    content_pub_ = create_publisher<std_msgs::msg::String>("truck_safety_info", 5);
+
+    // Setup service servers
+    inspection_request_service_server_ = create_service<std_srvs::srv::Trigger>("send_inspection_request",std::bind(&Node::inspectionRequestCallback, this, std_ph::_1, std_ph::_2, std_ph::_3));
+    
+    if(!this->safety_log_.empty()) {
+        std_msgs::msg::String msg_content;
+        msg_content.data = safety_log_;
+        content_pub_->publish(msg_content);
+    }
```

Comment:
This check and publication need to be occurring at a fixed frequency based off the timer callback in the ROS1 code